java/core/src/main/java/com/whylogs/api/writer/Writer.java

```diff
@@ -0,0 +1,17 @@
+package com.whylogs.api.writer;
+
+import java.util.Optional;
+
+// TODO: this is a temp holding class for logger that will be implmented next
+public abstract class Writer {
+  /*Validate an interval configuration for a given writer.
+  Some writer only accepts certain interval configuration. By default, this should return True for a valid
+  non-negative interval.*/
+  public void check_interval(int interval_seconds) {
```

Comment:
Why is this in the base class? (same question for Python)

Reply:
That's fair. I'm curious to that as well. I will explore that further in the next iteration. Right now it was just a holder